src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceChatClient.cs

```diff
@@ -273,109 +273,99 @@ private static ChatRole ToChatRole(global::Azure.AI.Inference.ChatRole role) =>
+        if (options.RawRepresentation is ChatCompletionsOptions azureAIOptions)
         {
-            result.FrequencyPenalty = options.FrequencyPenalty;
-            result.MaxTokens = options.MaxOutputTokens;
-            result.NucleusSamplingFactor = options.TopP;
-            result.PresencePenalty = options.PresencePenalty;
-            result.Temperature = options.Temperature;
-            result.Seed = options.Seed;
-
-            if (options.StopSequences is { Count: > 0 } stopSequences)
-            {
-                foreach (string stopSequence in stopSequences)
-                {
-                    result.StopSequences.Add(stopSequence);
-                }
-            }
+            result = azureAIOptions;
+            result.Messages = ToAzureAIInferenceChatMessages(chatContents).ToList();
+            result.Model = options.ModelId ?? _metadata.DefaultModelId ?? throw new InvalidOperationException("No model id was provided when either constructing the client or in the chat options.");
+        }
+        else
+        {
+            result = CreateAzureAIOptions(chatContents, options);
+        }
+
+        result.FrequencyPenalty = options.FrequencyPenalty;
+        result.MaxTokens = options.MaxOutputTokens;
+        result.NucleusSamplingFactor = options.TopP;
+        result.PresencePenalty = options.PresencePenalty;
+        result.Temperature = options.Temperature;
+        result.Seed = options.Seed;
 
-            // These properties are strongly typed on ChatOptions but not on ChatCompletionsOptions.
-            if (options.TopK is int topK)
+        if (options.StopSequences is { Count: > 0 } stopSequences)
+        {
+            foreach (string stopSequence in stopSequences)
             {
-                result.AdditionalProperties["top_k"] = new BinaryData(JsonSerializer.SerializeToUtf8Bytes(topK, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(int))));
+                result.StopSequences.Add(stopSequence);
             }
+        }
 
-            if (options.AdditionalProperties is { } props)
+        if (options.Tools is { Count: > 0 } tools)
+        {
+            foreach (AITool tool in tools)
             {
-                foreach (var prop in props)
+                if (tool is AIFunction af)
                 {
-                    switch (prop.Key)
-                    {
-                        // Propagate everything else to the ChatCompletionsOptions' AdditionalProperties.
-                        default:
-                            if (prop.Value is not null)
-                            {
-                                byte[] data = JsonSerializer.SerializeToUtf8Bytes(prop.Value, AIJsonUtilities.DefaultOptions.GetTypeInfo(typeof(object)));
-                                result.AdditionalProperties[prop.Key] = new BinaryData(data);
-                            }
-
-                            break;
-                    }
+                    result.Tools.Add(ToAzureAIChatTool(af));
```

Comment:
Tools and some other collections are initialized in the ctor, if the collection is not empty here, do we want to still `Add` elements specified in ChatOptions? I feel it would not align with the write-only-if-null semantic you described for the nullable properties of the underlying options, and it may lead to errors like duplicated elements if the user defined the tools in both the underlying and our options.

Reply:
Merging the tools from `ChatOptions.Tools` and `ChatOptions.RawRepresentation.ToolsOrWhatever` in a single collection is what I'd expect to happen.

If the developer duplicates entries across both sources that's their mistake and their bug to fix, but I don't think people will commonly have any way to make a tool show up in both places since they would be represented by different object types.

Reply:
> Merging the tools from ChatOptions.Tools and ChatOptions.RawRepresentation.ToolsOrWhatever in a single collection is what I'd expect to happen.

Me, too.